### PR TITLE
rune/libenclave/attestation: move sgx-epid-challenger.go to the sgx/sgx_challenger package

### PR DESCRIPTION
**Pull request by hustliyilin**
sgx-related codes should be located into rune/libenclave/attestation/sgx
package. More specifically, {sgx-attester-related, sgx-challenger-related,
sgx-verifier-related} should be placed into
rune/libenclave/attestation/{sgx_attester, sgx_challenger, sgx_verifier} packages.

Signed-off-by: Yilin Li <YiLin.Li@linux.alibaba.com>

### CODE REVIEW

File: rune/libenclave/attestation/sgx/challenger/sgx-epid-challenger.go [hunk 1]
```diff
@@ -1,7 +1,8 @@
-package attestation // import "github.com/inclavare-containers/rune/libenclave/attestation"
+package challenger // import "github.com/inclavare-containers/rune/libenclave/attestation/sgx/challenger"
```
Comment (jiazhang0) [inline rune/libenclave/attestation/sgx/challenger/sgx-epid-challenger.go:1]: Change the package name to sgx_challenger.

File: rune/libenclave/internal/runtime/pal/pal_linux.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"github.com/go-restruct/restruct"
 	"github.com/inclavare-containers/epm/pkg/epm-api/v1alpha1"
 	"github.com/inclavare-containers/rune/libenclave/attestation"
+	_ "github.com/inclavare-containers/rune/libenclave/attestation/sgx/challenger"
 	_ "github.com/inclavare-containers/rune/libenclave/attestation/sgx/ias"
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/pal_linux.go:10]: This line is still necessary?

File: rune/libenclave/internal/runtime/pal/pal_linux.go [hunk 3]
```diff
@@ -44,10 +45,10 @@ func (pal *enclaveRuntimePal) Init(args string, logLevel string) error {
 	no_epm := true
 	if !strings.Contains(args, "no-epm") {
 		no_epm = false
-	       /* enclaveinfo.Layout retrieves from /proc/pid/mmaps, in file
+		/* enclaveinfo.Layout retrieves from /proc/pid/mmaps, in file
 		* mmaps /dev/sgx/enclave mmaping address is sorted from low
 		* address to high one. So layout[0].Addr will be minimum.
-		*/
+		 */
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/pal_linux.go:51]: Plz move these modifications to another commit.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: rune/libenclave/internal/runtime/pal/pal_linux.go (hunk 2 above)
Comment (hustliyilin) [reply rune/libenclave/internal/runtime/pal/pal_linux.go:10]: > This line is still necessary?

Yes, because `sgx-epid-challenger.go` has a init function, other packages(sgx challengers) must import it to do it's init function.

Comment (jiazhang0) [reply rune/libenclave/internal/runtime/pal/pal_linux.go:10]: I mean the **ias** package is still required to be imported in this way?

Comment (hustliyilin) [reply rune/libenclave/internal/runtime/pal/pal_linux.go:10]: > I mean the **ias** package is still required to be imported in this way?

ok, **ias** package should be imported by rune/libenclave/attestation/sgx/sgx-challenger/sgx-epid-challenger.go, so it's not imported here.

File: rune/libenclave/internal/runtime/pal/pal_linux.go (hunk 3 above)
Comment (hustliyilin) [reply rune/libenclave/internal/runtime/pal/pal_linux.go:50]: > Plz move these modifications to another commit.

Done, thanks.

**Review by jiazhang0 — APPROVED** [review]
